server/service/hosts.go

```diff
@@ -545,22 +556,30 @@ func refetchHostEndpoint(ctx context.Context, request interface{}, svc fleet.Ser
 }
 
 func (svc *Service) RefetchHost(ctx context.Context, id uint) error {
-	if err := svc.authz.Authorize(ctx, &fleet.Host{}, fleet.ActionList); err != nil {
-		return err
+	// can be already authorized if coming from device auth token endpoint
```

Comment:
Should we move this check into `svc.authz.Authorize(...)` instead? it has access to the context.

Reply:
I did think about this, but we can't do that because in many cases we need to call Authorize more than once and actually run all those different checks (e.g. authorize that user can read, then load the entity, then authorize that user can write or read more data based on more information such as teams being loaded). As mentioned in my reply to Lucas' comment a bit lower in the code, I could wrap this in a call to `authzctx.IsAlreadyAuthenticated(ctx) bool` just to make it a bit clearer.

Reply:
s/Authenticated/Authorized/

Reply:
SGTM!